internal/rpc/otel_interceptor.go

```diff
@@ -0,0 +1,420 @@
+package rpc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"connectrpc.com/connect"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/propagation"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+	"go.opentelemetry.io/otel/trace"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/TBD54566975/ftl/internal/log"
+)
+
+const (
+	otelFtlRequestKey       = "ftl.requestKey"
+	otelFtlVerbRef          = "ftl.verb.ref"
+	otelFtlVerbModule       = "ftl.verb.module"
```

Comment:
nittiest of nits: Jon appended `Attribute` to each of these variable names so I started doing that too. Kinda wondered about just doing `Attr` instead of brevity but either works